File: VB/XtraForm1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace EUD_NewReport
	Partial Public Class ReportChooser
		Inherits DevExpress.XtraEditors.XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub


        Private _index As Integer = -1

        Public Property Index() As Integer
            Get
                Return Me._index
            End Get
            Set(ByVal value As Integer)
                Me._index = value
            End Set
        End Property

        Private Sub ReportChooser_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Me.listBoxControl1.SelectedIndex = 0
        End Sub

        Private Sub listBoxControl1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles listBoxControl1.SelectedIndexChanged
            Me._index = Me.listBoxControl1.SelectedIndex
        End Sub

        Private Sub OKButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles listBoxControl1.DoubleClick, OKButton.Click
            Me.Close()
        End Sub

        Private Sub CancButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles CancButton.Click
            Me._index = -1
            Me.Close()
        End Sub
	End Class
End Namespace

File: CS/XtraForm1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace EUD_NewReport {
    public partial class ReportChooser : DevExpress.XtraEditors.XtraForm {
        public ReportChooser() {
            InitializeComponent();
        }

        private int index = -1;

        public int Index {
            get { return this.index; }
            set { this.index = value; }
        }

        private void ReportChooser_Load(object sender, EventArgs e) {
            this.listBoxControl1.SelectedIndex = 0;
        }

        private void listBoxControl1_SelectedIndexChanged(object sender, EventArgs e) {
            this.index = this.listBoxControl1.SelectedIndex;
        }

        private void OKButton_Click(object sender, EventArgs e) {
            this.Close();
        }

        private void CancButton_Click(object sender, EventArgs e) {
            this.index = -1;
            this.Close();
        }
    }
}